pkg/apps/strategy/rolling/rolling.go

```diff
@@ -220,13 +228,24 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	//
 	// Related upstream issue:
 	// https://github.com/kubernetes/kubernetes/pull/7183
-	to.Spec.Replicas = 1
+	oneReplica := int32(1)
+	to.Spec.Replicas = &oneReplica
+
+	// TODO: The upstream rolling updater should move to versioned API
```

Comment:
we can't get rid of the internal client objects entirely because upstream is still on internal here, but converting here once is still better that converting on every single request elsewhere.

Reply:
@mfojtik @tnozicka you are aware that `kubectl rolling-update` and all its related code is marked deprecated in 1.11, which means it'll be gone sometime in 1.14 or so. We should switch to the code behind `kubectl rollout` instead.

Reply:
That's why I was pointing it to you, it just needs to happen at some point in time, the sooner the better.

Reply:
I am not aware and I think this will be bad for us :/

@soltysh would you mind elaborating on how `kubectl rollout` replaces the functionality that `kubectl rolling-update` had for RC/RS?

Reply:
@soltysh I didn't know and I don't think there is any equivalent code in `kubectl rollout` as that's handled server side and for RS only.

Also this command is a base building block for custom strategies.